### PR TITLE
Export vault to HTML

### LINKED ISSUE #954: Suggestions for exporting features
Hello developer.
Is it possible to add a web format (html) to the format of the export function? The web page contains the "Issuer, Username, QR Code" information for each item.

### PR DESCRIPTION
**Pull request by CristianAUnisa**
Closes #954
I'm marking this as a draft because I'd like to ask if there's a preferred layout or style for the output. This is what I'm getting (zoom decreased to 50%), but it's pretty easy to adjust:

I'm going to create the test unit when I have more time to focus, I'm going to add more fields for now

Edit: would a table structure be better?

### CODE REVIEW

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -105,6 +105,7 @@
     <string name="export_warning_password">Exports are encrypted using a separate password configured in Security settings.</string>
     <string name="export_format_aegis">Aegis (.JSON)</string>
     <string name="export_format_google_auth_uri">Text file (.TXT)</string>
+    <string name="export_format_html">Web Page (.HTML)</string>
```
Comment (alexbakker) [inline app/src/main/res/values/strings.xml:108]: Nit: "Web page (.HTML)"

**Review by alexbakker — CHANGES_REQUESTED** [review]
Thanks for picking this up! The table structure looks good to me. I don't think we should go for the beauty prize, but perhaps the values in the cells could use a little padding and centered placement.

**Comment by CristianAUnisa** [comment]
Thank you for the answer. I forgot to ask but an importer is needed too, doesn't it?

**Comment by alexbakker** [comment]
I don't think we should have an importer for this actually. This is just for people who'd like to have a backup in a format that's easy to print.

**Comment by CristianAUnisa** [comment]
Padding set and text centered; I changed the html internal title too. This is something made quickly, but if it needs to look nicer, just let me know (like showing the Favorite value as a checkbox).
I think an importer is needed for creating a test, but I'm going to check if I can put something together without making a mess.

**Comment by alexbakker** [comment]
Looks good! I wouldn't worry to much about the test. If you want, you can add a basic one where you just run the export and check that it doesn't break. I wouldn't bother with reading the file back, as that's likely going to get messy, like you said.

Now that I've taken a closer look at this, I actually have a couple of extra suggestions:
- Instead of using a ``StringBuilder``, wrap the ``OutputStream`` in a ``PrintStream`` so that we don't have to keep the whole file in-memory.
- Add an ``h1`` to the top of the page with the same "Aegis Authenticator Export" title.

I'd also be happy to merge this as-is and make these small modifications myself afterwards.

**Comment by CristianAUnisa** [comment]
I'm not knowledgeable about what's needed for Yandex and mOTP, I'm going to take a look but if you want to merge this, I applied the other three suggestions and I decided to make the "Aegis Authenticator Export" a string resource

**Pull request merged**